MIPs/mip-x.md

```diff
@@ -0,0 +1,218 @@
+
+  ```markdown
+  validateParams(fromToken[0], [address], 'fromToken');
+  validateParams(toToken, [address], 'toToken');
+  ```
+
+- **Invalid User Address**: If a non-existent address is provided in `userAddress`, then the method will throw an error 'This address does not exist'. See more details about error codes in the api-spec: https://github.com/MetaMask/api-specs/pull/201
+
+  ```markdown
+  if (!dappConnectedAccount) {
+  throw rpcErrors.invalidParams('This address does not exist');
+  }
+  ```
+
+- **Unsupported Multiple Tokens Swap**: As of now, it supports only a single token swap. In case of multiple tokens provided in the `fromToken` object, an 'Currently we do not support multiple tokens swap' error is thrown. See more details about error codes in the api-spec: https://github.com/MetaMask/api-specs/pull/201
+
+  ```markdown
+  if (fromToken.length > 1) {
+  throw rpcErrors.invalidParams(
+  'Currently we do not support multiple tokens swap',
+  );
+  }
+  ```
+
+- **Inactive or Unsupported Swaps**: If the swap is inactive or is not possible on the current chain, an alert with the message 'Swap is not active or is not possible on this chain' will be triggered. See more details about error codes in the api-spec: https://github.com/MetaMask/api-specs/pull/201
+
+  ```markdown
+  if (!isSwappable) {
+  Alert.alert(`Swap is not available on this chain ${networkName}`);
+  throw rpcErrors.methodNotSupported(
+  `Swap is not available on this chain ${networkName}`,
+  );
+  }
+  ```
+
+- **Cross-Chain Swap Attempt**: Given the current limitations, MetaMask does not support cross-chain swaps. If a swap request involves tokens from different blockchains (as indicated by their CAIP-10 formatted addresses), MetaMask will throw an error to indicate this unsupported operation. This ensures developers and users are aware of the limitations and can adjust their actions accordingly. See more details about error codes in the api-spec: https://github.com/MetaMask/api-specs/pull/201
+
+  ```markdown
+  if (parseChainIdFromCaip10(fromToken.address) !== parseChainIdFromCaip10(toToken.address)) {
+  throw rpcErrors.methodNotSupported(
+  'Cross-chain swaps are currently not supported. Both fromToken and toToken must be on the same blockchain.',
+  );
+  }
+  ```
+
+- **Invalid caip-10 address**: If the dapp provides an invalid caip-10 address we throw an error, that could be an invalid address for the account address or tokens addresses
+
+```markdown
+try {
+parsedCaip10UserAddress = parseCaip10Address(user_address);
+} catch (error) {
+throw rpcErrors.invalidParams('Invalid caip-10 user address');
+}
+
+try {
+parsedCaip10FromTokenAddress = parseCaip10Address(
+fromToken[0].address,
+);
+} catch (error) {
+throw rpcErrors.invalidParams('Invalid caip-10 fromToken address');
+}
+
+try {
+parsedCaip10ToTokenAddress = parseCaip10Address(toToken.address);
+} catch (error) {
+throw rpcErrors.invalidParams('Invalid caip-10 toToken address');
```

Comment:
Maybe same here: `caip-10` -> `CAIP-10`?  I know these are the actual error messages so maybe not...?

Reply:
I think this section would be clearer in english instead of a try catch code example